### PR TITLE
is_resource: adjust type specification for truthy context only

### PR DESCRIPTION
**Pull request by janedbal**
Should fix https://github.com/phpstan/phpstan/issues/9687

### CODE REVIEW

**Comment by thg2k** [comment]
I don't think you need a dedicated class for this, just replace the stub assertion with:

```
@phpstan-assert-if-true =resource $value
```

The (undocumented?) equal sign indicates that if false the opposite should not be asserted.

**Comment by janedbal** [comment]
Oh god, the whole psalm-originated `@phpstan-assert-if-true` is so un-intuitive! 

Thank you!

File: tests/PHPStan/Analyser/data/specifiedTypesUsingIsFunctions.php [hunk 1]
```diff
@@ -41,7 +41,8 @@ function () {
 		throw new \Exception();
 	}
 
-	if (!is_resource($resource)) {
+	$resource = fopen('php://memory', 'r');
+	if ($resource === false) {
```
Comment (ondrejmirtes) [inline tests/PHPStan/Analyser/data/specifiedTypesUsingIsFunctions.php]: Not sure why the change is here. Usually changing existing (and especially this old) tests is a red flag. Did we break something?

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]
Since we're trying to fix "Call to function is_resource() with resource will always evaluate to true.", we need a test for ImpossibleCheckTypeFunctionCallRule. Thanks.

File: tests/PHPStan/Analyser/data/specifiedTypesUsingIsFunctions.php (hunk 1 above)
Comment (janedbal) [reply tests/PHPStan/Analyser/data/specifiedTypesUsingIsFunctions.php]: Reverted, not needed.

**Comment by ondrejmirtes** [comment]
Thank you!
